jans-cedarling/cedarling/src/jwt/token.rs

```diff
@@ -0,0 +1,264 @@
+    {
+        let claims = HashMap::<String, Value>::deserialize(deserializer)?;
+        Ok(Self { claims, iss: None })
+    }
+}
+
+impl From<HashMap<String, Value>> for TokenData<'_> {
+    fn from(claims: HashMap<String, Value>) -> Self {
+        Self { claims, iss: None }
+    }
+}
+
+impl<'a> TokenData<'a> {
+    pub fn new(claims: HashMap<String, serde_json::Value>, iss: Option<&'a TrustedIssuer>) -> Self {
+        Self { claims, iss }
+    }
+
+    pub fn from_json_map(map: serde_json::Map<String, serde_json::Value>) -> Self {
+        Self::new(HashMap::from_iter(map), None)
+    }
+
+    pub fn has_claim(&self, name: &str) -> bool {
+        self.claims.contains_key(name)
+    }
+
+    pub fn get_claim(&self, name: &str) -> Option<TokenClaim> {
+        self.claims.get(name).map(|value| TokenClaim {
+            key: name.to_string(),
+            value,
+        })
+    }
+
+    pub fn get_logging_info(&'a self, claim: &'a str) -> HashMap<&'a str, &'a serde_json::Value> {
+        let claim = if !claim.is_empty() { claim } else { "jti" };
+
+        let iter = [self.claims.get(claim).map(|value| (claim, value))]
+            .into_iter()
+            .flatten();
+
+        HashMap::from_iter(iter)
+    }
+}
+
+pub struct TokenClaim<'a> {
+    key: String,
+    value: &'a serde_json::Value,
+}
+
+impl TokenClaim<'_> {
+    pub fn key(&self) -> &str {
+        &self.key
+    }
+
+    pub fn value(&self) -> &serde_json::Value {
+        self.value
+    }
+
+    pub fn as_i64(&self) -> Result<i64, TokenClaimTypeError> {
+        self.value
+            .as_i64()
+            .ok_or(TokenClaimTypeError::type_mismatch(
+                &self.key, "i64", self.value,
+            ))
+    }
+
+    pub fn as_str(&self) -> Result<&str, TokenClaimTypeError> {
+        self.value
+            .as_str()
+            .ok_or(TokenClaimTypeError::type_mismatch(
+                &self.key, "String", self.value,
+            ))
+    }
+
+    pub fn as_bool(&self) -> Result<bool, TokenClaimTypeError> {
+        self.value
+            .as_bool()
+            .ok_or(TokenClaimTypeError::type_mismatch(
+                &self.key, "bool", self.value,
+            ))
+    }
+
+    pub fn as_array(&self) -> Result<Vec<TokenClaim>, TokenClaimTypeError> {
+        self.value
+            .as_array()
+            .map(|array| {
+                array
+                    .iter()
+                    .enumerate()
+                    .map(|(i, v)| TokenClaim {
+                        // show current key and index in array
+                        key: format!("{}[{}]", self.key, i),
+                        value: v,
+                    })
+                    .collect()
+            })
+            .ok_or(TokenClaimTypeError::type_mismatch(
+                &self.key, "Array", self.value,
+            ))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("Type mismatch for key '{key}': expected: '{expected_type}', but found: '{actual_type}'")]
```

Comment:
Error message should not start with upper case letter. 
For example look to the https://docs.rs/thiserror/latest/thiserror/

Reply:
changed in [e170008](https://github.com/JanssenProject/jans/pull/10436/commits/e1700083ce09fb26f1093c0324a797e73855fd55)